src/main/ruby/core/truffle/cext.rb

```diff
@@ -44,10 +50,33 @@ def self.main(argv = ARGV)
           end
         end
       end
-
+      libraries = resolve_libraries(libraries, search_paths)
       Truffle::CExt.linker(output, libraries, files)
     end
 
+    def self.resolve_libraries(libraries, search_paths)
+      require 'pathname'
+      resolved_libraries = []
+      libraries.each do |lib|
+        if Pathname.new(lib).absolute?
```

Comment:
It's quite unfortunate there is no `File.absolute?` and loading pathname just for that.
But it's clearer than checking with `start_with?('/')` or `File.absolute_path(lib) == lib`.